pkg/asset/manifests/ovirt/cloudproviderconfig.go

```diff
@@ -0,0 +1,26 @@
+package ovirt
+
+// CloudProviderConfig represents ovirt's cloud provider config map
+type CloudProviderConfig struct {
+	URL               string `json:"ovirt_url,omitempty"`
+	Username          string `json:"ovirt_username,omitempty"`
+	Password          string `json:"ovirt_password,omitempty"`
+	Cafile            string `json:"ovirt_cafile,omitempty"`
+	Insecure          bool   `json:"ovirt_insecure,omitempty"`
+	StorageDomainName string `json:"storage_domain_name,omitempty"`
+	ClusterID         string `json:"cluster_id,omitempty"`
+	TemplateID        string `json:"template_id,omitempty"`
+}
+
+// NewCloudProviderConfig creates a new CloudProviderConfig
+func NewCloudProviderConfig(URL string, username string, password string, cafile string, insecure bool, storageDomainName string, clusterID string, templateID string) *CloudProviderConfig {
```

Comment:
most of the platforms provide a serialized form of the cloud-provider-config, so that it can be directly set in the Configmap.

Reply:
What is the role of the cloudprovider in the cluster? ovirt cloud provider can only extarct IP from nodes. Is this its usage? Maybe this is not needed for this platform at all.

Reply:
Not all cloudproviders (for example, AWS) require a cloud provider config. It sounds like oVirt may not... I think the cloud provider config is used to provide config for [cloud providers](https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/) using the [cloud controller manager](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/). For example, GCP uses the cloud provider config to create the ingress LB. I hope that helps.